packages/template-retail-react-app/app/components/link/index.jsx

```diff
@@ -8,20 +8,16 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import {Link as ChakraLink} from '@chakra-ui/react'
 import {Link as SPALink, NavLink as NavSPALink} from 'react-router-dom'
-import {buildPathWithUrlConfig} from '../../utils/url'
 import useSite from '../../hooks/use-site'
 import useLocale from '../../hooks/use-locale'
+import useAppConfig from '../../hooks/use-app-config'
 
 const Link = React.forwardRef(({href, to, useNavLink = false, ...props}, ref) => {
     const _href = to || href
-    const site = useSite()
-    const locale = useLocale()
-
-    // if alias is not defined, use site id
-    const updatedHref = buildPathWithUrlConfig(_href, {
-        locale: locale.alias || locale.id,
-        site: site.alias || site.id
-    })
+    const {site} = useSite()
+    const {locale} = useLocale()
+    const appConfig = useAppConfig()
+    const updatedHref = appConfig.urlTemplateLiteral(_href, site.alias || site.id, locale.id)
```

Comment:
What if we have a hook that makes it easier to get the `urlTemplateLiteral`?
```suggestion
    const urlTemplateLiteral = useUrlTemplateLiteral()
    const updatedHref = urlTemplateLiteral(_href, site.alias || site.id, locale.id)
```

Reply:
That's not necessary, you can accomplish the same thing by just destructuring:
```suggestion
    const {urlTemplateLiteral} = useAppConfig()
    const updatedHref = urlTemplateLiteral(_href, site.alias || site.id, locale.id)
```

Reply:
Sorry, that's right. But what I meant is more like "do we need the concept of app config object that has `urlTemplateLiteral`?" Based on the implementation right now, it seems like what we want is simply just the `urlTemplateLiteral`.